Update conventions.rst (#6333)
cakephp/docs: en/intro/conventions.rst

maintenance nightmare of tracking config files. Conventions also make for a very
uniform development experience, allowing other developers to jump in and help.

.. note::

>     CamelCase is the practice of writing compound words or phrases so that each word or abbreviation in the middle of the sentence begins with an uppercase letter.

Reviewer: "in the middle" no, every word

- CamelCase
- camelBacked

see Inflector and other sources here.

>     CamelCase is the practice of writing compound words or phrases so that each word or abbreviation in the middle of the sentence begins with an uppercase letter.

Reviewer 2: ```suggestion
    Camel case is the practice of writing compound words or phrases so that each word other than the first begins with an uppercase letter. E.g. ``redApple``
```

>     PascalCase is the practice of writing compound words or phrases so that each word or abbreviation begins with an uppercase letter.

Reviewer 2: ```suggestion
    Pascal case is the practice of writing compound words or phrases so that each word including the first begins with an uppercase letter. E.g. ``RedApple``.
```

>     Both do not use spaces or punctuation.

Reviewer 2: ```suggestion
    Both camel case and pascal case do not use spaces or punctuation.
```

conventions for the url array::

    $this->Html->link('link-title', [
        'prefix' => 'MyPrefix' // PascalCase
        'plugin' => 'MyPlugin', // PascalCase
        'controller' => 'ControllerName', // PascalCase

>         'action' => 'actionName' // camelCase

Reviewer 3: This is not camelCase, it's camelBacked. CamelCase is same as PascalCase

Reviewer: I would also prefer if we used CamelCase since that is everywhere else as such.

Author: > This is not camelCase, it's camelBacked. CamelCase is same as PascalCase

https://medium.com/better-programming/string-case-styles-camel-pascal-snake-and-kebab-case-981407998841

and other:

![image](https://user-images.githubusercontent.com/50879411/71027754-a3288800-20ea-11ea-8b9a-a7bf16d123b4.png)

"CamelBacked" is variation and synonym.    https://en.wikipedia.org/wiki/Camel_case

thanks you

    rules. See the documentation about :doc:`/core-libraries/inflector` for more
    information.

Helpers
=======

> Helper class files should be put in src/View/Helper. For example: src/View/Helper/LinkHelper.php

Reviewer 2: ```suggestion
Helper class files should be put in **src/View/Helper**. For example: **src/View/Helper/LinkHelper.php**
```

Paths are bolded in the book.

Reviewer: Directories should have trailing slash.

> Helper classes should be suffixed with Helper, in PascalCase. For example: "LinkHelper", filename: "link.php", and "GoogleHelper" filename: "google.php".

Reviewer: The filenames are not correct here, those would violate PSR naming.

> Helper classes should be suffixed with Helper, in PascalCase. For example: "LinkHelper", filename: "link.php", and "GoogleHelper" filename: "google.php".

Reviewer 2: ```suggestion
Helper classes should be suffixed with Helper, in PascalCase. For example: ``LinkHelper`` would be in a file named **LinkHelper.php**, and ``GoogleHelper`` would be in a file named **GoogleHelper.php**.
```

> When referencing helper class names you should omit the Helper suffix. For example: $this->loadHelper('Link') or $this->loadHelper('Google');

Reviewer 2: ```suggestion
When referencing helper class names you should omit the Helper suffix. For example: ``$this->loadHelper('Link')`` or ``$this->loadHelper('Google');``
```

<https://github.com/FriendsOfCake/awesome-cakephp/blob/master/CONTRIBUTING.md#tips-for-creating-cakephp-plugins>`__
for details.

Constants
=========

> Always keep the same pattern, as a suggestion, in capital letters and underlines:

Reviewer 3: s/underline/underscore

> Always keep the same pattern, as a suggestion, in capital letters and underlines:

Reviewer 2: ```suggestion
Always keep the same pattern, as a suggestion, in capital letters and underscore:
```

> "MIN_USER_AGE", "MY_CONSTANT_PERSONAL".

Reviewer 2: ```suggestion
``MIN_USER_AGE``, ``MY_CONSTANT_PERSONAL``.
```

Variables
=========

> Always use the same pattern throughout the camelCase application as in $userComment, $total, $countPerson.

Reviewer 2: ```suggestion
Always use the same pattern throughout the camelCase application as in ``$userComment``, ``$total``, ``$countPerson``.
```

> The first letter is uppercase if the variable represents an object:

Reviewer 3: We no longer follow this convention in 3.x+.

Reviewer: To clarify: This is only used still where the object is a special locator based class attribute

$this->loadModel('Foo') and $this->Foo

but

$foo = new Foo();

etc.
